### ISSUE TITLE
nested ensure misdetected as not coming first.

### ISSUE THREAD

**Issue by jamesooden** [rodjek/puppet-lint#410]
Taking the following puppet code:

```
class thing {
    class {'thang':
        stuff => {
            'stuffing' => {
                ensure => 'present',
                blah   => 'bleah',
            }
        },
    }
}
```

And running puppet-lint against it I will get the following warning:

```
WARNING: ensure found on line but it's not the first attribute on line 6
```

Which is not true, as the ensure is embedded in the next level of the structure and at that level it is the first attribute.

**Comment by ghoneycutt** [comment]
ensure is not an attribute at all in this case, it is a key in a hash.

**Comment by jamesooden** [comment]
Right, and should not cause the warning to be flagged.

**Comment by benohara** [comment]
We see this for stuff relating to the mysql module, users, grants etc :(

**Comment by jeff1evesque** [comment]
@jamesooden :+1:

**Comment by jeff1evesque** [comment]
I may look for alternative linters, as I find this a necessity.  Otherwise, my travis will give me a nice red button on my `README.md` stating that `puppet-lint` is unable to detect `ensure` properly.

**Comment by ghoneycutt** [comment]
@jeff1evesque You can disable checks.

Here is an example

https://github.com/ghoneycutt/puppet-module-vim/blob/master/Rakefile#L3

**Comment by jeff1evesque** [comment]
@ghoneycutt, thank you!  I will most likely consider the following sytnax:

- `puppet-lint --no-ensure_first_param-check path/to/file.pp`

**Comment by rnelson0** [comment]
@jamesooden @benohara Can either of you link to a use of this in practical code? The example code does not make sense, as `ensure` should be a quoted word where it's being used as a hash key.

**Comment by jamesooden** [comment]
Not anymore; I don't work for that company any more.   I assure you it was
in practical code.   Sorry the example doesn't make sense to you...James

**Comment by rnelson0** [comment]
@jamesooden I wasn't seeing the context properly, but I see it now. Working on a PR.

**Comment by rnelson0** [comment]
Fixed in #490. Thanks!

**Comment by Fabian1976** [comment]
Is this allready in a release? Because with the most recent version of lint (2.3.6), I still get this error:

```manifests/profile/citrix/vda.pp - WARNING: ensure found on line but it's not the first attribute on line 47```

This is the piece of code:
```
  dsc { 'xWindowsFeature_RDS-RD-Server':
    notify                  => Reboot['After_Features'],
    dsc_resource_name       => 'xWindowsFeature',
    dsc_resource_module     => 'xPSDesiredStateConfiguration',
    dsc_resource_properties => {
      ensure               => 'present',
      name                 => 'RDS-RD-Server',
      psdscrunascredential => {
        dsc_type       => 'MSFT_Credential',
        dsc_properties => $admin,
      },
    },
  }
```

**Comment by rodjek** [comment]
Confirmed that this seems to have regressed.
